### PR TITLE
bpo-45731: Handle --enable-loadable-sqlite-extensions in configure (GH-29434)

### PR DESCRIPTION
**Pull request by tiran**
Signed-off-by: Christian Heimes <christian@python.org>

https://bugs.python.org/issue45731

Automerge-Triggered-By: GH:tiran

### CODE REVIEW

**Comment by tiran** [comment]
The extra noise in configure will go away after GH-29433 is merged.

File: configure.ac [hunk 1]
```diff
@@ -3199,10 +3199,14 @@ AC_ARG_ENABLE(loadable-sqlite-extensions,
...
                              [support loadable extensions in _sqlite module, see Doc/library/sqlite3.rst (default is no)]),
               [],
-              [enable_loadable_sqlite_extensions="no"])
-
+              [enable_loadable_sqlite_extensions=no])
 AC_MSG_RESULT($enable_loadable_sqlite_extensions)
 
+AS_VAR_IF([enable_loadable_sqlite_extensions], [yes], [
+  AC_DEFINE(PY_SQLITE_ENABLE_LOADABLE_EXTENSION, 1,
+  [Define to 1 to build sqlite module with loadable extensions.])
```
Comment (erlend-aasland) [inline configure.ac]: Nitpick:
```suggestion
  [Define to 1 to build the sqlite3 module with loadable extensions support.])
```

File: configure.ac [hunk 2]
```diff
@@ -3199,10 +3199,14 @@ AC_ARG_ENABLE(loadable-sqlite-extensions,
               AS_HELP_STRING([--enable-loadable-sqlite-extensions],
                              [support loadable extensions in _sqlite module, see Doc/library/sqlite3.rst (default is no)]),
               [],
-              [enable_loadable_sqlite_extensions="no"])
-
+              [enable_loadable_sqlite_extensions=no])
 AC_MSG_RESULT($enable_loadable_sqlite_extensions)
 
+AS_VAR_IF([enable_loadable_sqlite_extensions], [yes], [
+  AC_DEFINE(PY_SQLITE_ENABLE_LOADABLE_EXTENSION, 1,
```
Comment (erlend-aasland) [inline configure.ac]: Nitpick :) I would prefer `PY_SQLITE_ENABLE_LOADABLE_EXTENSIONS` (with the plural S) or `PY_SQLITE_ENABLE_LOAD_EXTENSION`. Perhaps the latter, as it is shorter.

File: configure.ac (hunk 1 above)
Comment (erlend-aasland) [reply configure.ac]: BTW, are we using `AC_CONFIG_HEADERS` somewhere?

**Review by erlend-aasland — APPROVED** [review]
LGTM! Left some super nitpicky comments.

File: configure.ac (hunk 1 above)
Comment (tiran) [reply configure.ac]: yes, configure.ac has ``AC_CONFIG_HEADER(pyconfig.h)``

Comment (erlend-aasland) [reply configure.ac]: Is that in another PR? I can't find it in this checkout.

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**